tests/test_pipelines_common.py

```diff
@@ -120,15 +121,17 @@ def test_pipeline_call_implements_required_args(self):
             if param == "kwargs":
                 # kwargs can be added if arguments of pipeline call function are deprecated
                 continue
-            assert param in ALLOWED_REQUIRED_ARGS
+            assert param in self.allowed_required_args
 
         optional_parameters = set({k for k, v in parameters.items() if v.default != inspect._empty})
 
-        required_optional_params = ["generator", "num_inference_steps", "return_dict"]
-        for param in required_optional_params:
+        for param in self.required_optional_params:
             assert param in optional_parameters
 
     def test_inference_batch_consistent(self):
+        self._test_inference_batch_consistent()
+
+    def _test_inference_batch_consistent(self, batch_sizes=[2, 4, 13]):
         components = self.get_dummy_components()
         pipe = self.pipeline_class(**components)
```

Comment:
For tests that needed arguments to be tweaked, I put the main test body in a `_` prefixed method which the subclass can call with modified arguments. Lemme know if that's an acceptable pattern!